### PR TITLE
Enhance data processing and deps

### PR DESCRIPTION
**Pull request by colinfrederickson**
1. **Employee ID Verification**:
   - Enhanced the `checkApiForExistingWorkers` function to improve the accuracy of employee ID matching. This was achieved by ensuring consistent data types during comparison and adding detailed logging for troubleshooting potential discrepancies.
   - Provided clearer error messaging to help identify issues related to data mismatches or API call failures.

2. **Addition of File Extractors**:
   - Introduced new extractors to handle various file formats including XLSX, JSON, XML, ZIP, and pipe-delimited TXT. This allows for a more diverse range of input data formats, improving the flexibility and usability of our system.

3. **Package Version Updates**:

    "@flatfile/api": "1.5.22",
    "@flatfile/configure": "0.5.38",
    "@flatfile/hooks": "1.3.2",
    "@flatfile/listener": "0.3.15",
    "@flatfile/plugin-automap": "^0.0.3",
    "@flatfile/plugin-delimiter-extractor": "^0.5.0",
    "@flatfile/plugin-json-extractor": "^0.5.0",
    "@flatfile/plugin-record-hook": "1.0.4",
    "@flatfile/plugin-xlsx-extractor": "1.6.0",
    "@flatfile/plugin-xml-extractor": "^0.4.0",
    "@flatfile/plugin-zip-extractor": "^0.3.4",
"flatfile": "3.5.8",

These updates aim to streamline the data processing workflow, reduce errors, and enhance the user experience by supporting a broader range of data input formats.

### CODE REVIEW

File: recordHooks/employees/apiValidation.ts [hunk 1]
```diff
@@ -1,30 +1,32 @@
...
+      employees
+        .slice(0, 5)
+        .map((e) => ({ id: e.employeeId, type: typeof e.employeeId }))
+    );
 
-    // Check if the Applicant_ID matches an id from the API data
+    console.log('Employee_Id:', employeeId);
     const matchingEmployee = employees.find((employee) => {
-      return employee.employeeId === employeeId;
+      return String(employee.employeeId) === String(employeeId); // Convert both to strings for comparison
```
Comment (johnmosesman) [inline recordHooks/employees/apiValidation.ts:10]: When are there different types here? Also you could `.toString()` on them if you wanted to make sure they are both strings

Comment (colinfrederickson) [reply recordHooks/employees/apiValidation.ts:10]: it's super weird, upon import of data the types appear to be different and no error appears on the record.

When data is updated, the types are the same and an error appears on the record.

This fix solved the initial upload issue.

Comment (johnmosesman) [reply recordHooks/employees/apiValidation.ts:10]: following up from our sync today: it's weird how this is a number for some values, but in the meantime casting and comparing is fine.

File: recordHooks/employees/apiValidation.ts [hunk 2]
```diff
@@ -1,30 +1,32 @@
...
+    console.log('Employee_Id Type:', typeof employeeId); // Log the type of employeeId
 
-    console.log('Employee_Id:', employeeId); // Log the current value of Applicant_ID
+    // Log the first few employees' IDs and their types for comparison
+    console.log(
+      'Sample Employee IDs and their types:',
+      employees
+        .slice(0, 5)
+        .map((e) => ({ id: e.employeeId, type: typeof e.employeeId }))
+    );
```
Comment (johnmosesman) [inline recordHooks/employees/apiValidation.ts]: Can you cut or comment out these logs?

**Review by johnmosesman — APPROVED** [review]

**Pull request merged**